includes/class-amp-theme-support.php

```diff
@@ -2495,6 +2503,140 @@ public static function enqueue_assets() {
 		wp_styles()->add_data( 'amp-default', 'rtl', 'replace' );
 	}
 
+	/**
+	 * Setup pages to have the paired browsing client script so that the app can interact with it.
+	 *
+	 * @since 1.5.0
+	 *
+	 * @return void
+	 */
+	public static function setup_paired_browsing_client() {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET[ self::PAIRED_BROWSING_QUERY_VAR ] ) ) {
+			return;
+		}
+
+		$asset_file   = AMP__DIR__ . '/assets/js/amp-paired-browsing-client.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
+
+		wp_enqueue_script(
+			'amp-paired-browsing-client',
+			amp_get_asset_url( '/js/amp-paired-browsing-client.js' ),
+			$dependencies,
+			$version,
+			true
+		);
+
+		// Force dev mode to be enabled. This ensures that the enqueued script and its dependencies
+		// will be present when the admin bar is not showing.
+		add_filter( 'amp_dev_mode_enabled', '__return_true' );
```

Comment:
@pierlon I just realized that this makes every page invalid AMP, because every page is forced to be in dev mode and the custom script is enqueued. Two options I see:

1. Short-circuit this function if an user is not logged-in (or rather, if `amp_is_dev_mode()` returns `false`). Since search crawlers will never be logged-in, they will never then see an invalid AMP page. This would also require entire paired browsing app should `wp_die()` if the user is not logged in.
2. Go back to only including this JS if the admin bar is showing. Accessing paired browsing here would still need to include the `wp_die()` if `! amp_is_dev_mode()`.

Reply:
See #3863.